addons/cookoff/functions/fnc_handleDamageBox.sqf

```diff
@@ -1,72 +1,74 @@
 #include "..\script_component.hpp"
 /*
  * Author: KoffeinFlummi, commy2
- * Handles all incoming damage for boxi
+ * Handles all incoming damage for boxes.
  *
  * Arguments:
  * HandleDamage EH <ARRAY>
  *
  * Return Value:
- * Damage to be inflicted. <NUMBER>
+ * Damage to be inflicted (can be nil) <NUMBER>
  *
  * Example:
  * _this call ace_cookoff_fnc_handleDamageBox
  *
  * Public: No
  */
 
-params ["_vehicle", "", "_damage", "_source", "_ammo", "_hitIndex", "_shooter"];
+params ["_box", "", "_damage", "_source", "_ammo", "_hitIndex", "_instigator"];
 
-// it's already dead, who cares?
-if (damage _vehicle >= 1) exitWith {};
+if (!local _box) exitWith {};
 
-// If cookoff is disabled exit
-if (_vehicle getVariable [QGVAR(enable), GVAR(enable)] in [0, false]) exitWith {};
+// If it's already dead, ignore
+if (damage _box >= 1) exitWith {};
 
-// get hitpoint name
+// If cookoff for boxes is disabled, exit
+if (!GVAR(enableAmmobox) || {GVAR(ammoCookoffDuration) == 0}) exitWith {};
+
+if !(_box getVariable [QGVAR(enableAmmoCookoff), true]) exitWith {};
+
+// Get hitpoint name
 private _hitpoint = "#structural";
 
 if (_hitIndex != -1) then {
-    _hitpoint = toLower ((getAllHitPointsDamage _vehicle param [0, []]) select _hitIndex);
+    _hitpoint = ((getAllHitPointsDamage _box) param [0, []]) select _hitIndex;
 };
 
-// get change in damage
-private _oldDamage = 0;
+if !(_hitpoint == "#structural" && {_damage > 0.5}) exitWith {};
```

Comment:
Is this correct? Why do we care about `_hitIndex` above then?

Reply:
I pretty much left this code alone - I'll take a more in depth look at it though, as, admittedly, it's not very intuitive.

Reply:
Latest commit should address this, although I need to retest it.

Reply:
I have retested and it's all good. Latest commit also includes some minor tweaks.